server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -1203,9 +1204,10 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
          * @param fieldNode         the root node of the map of mappings for this field
          */
         public final void parse(String name, MappingParserContext parserContext, Map<String, Object> fieldNode) {
-            Map<String, Parameter<?>> paramsMap = new HashMap<>();
+            final Parameter<?>[] params = getParameters();
+            Map<String, Parameter<?>> paramsMap = Maps.newHashMapWithExpectedSize(params.length);
             Map<String, Parameter<?>> deprecatedParamsMap = new HashMap<>();
```

Comment:
the params map is hotter than the deprecated params map, because many params are added to it? Leave a comment around this to explain why we create the map this way?

Reply:
Yea, the deprecated names are mostly empty I think + we don't know the size up front without iterating all parameters -> added comment about that.